code/game/objects/structures/window.dm

```diff
@@ -175,67 +175,99 @@
+				setAnchored(!anchored)
+				to_chat(user, "<span class='notice'>You [anchored ? "fasten the window to":"unfasten the window from"] the floor.</span>")
+			return TRUE
+	else
+		return FALSE
 
+/obj/structure/window/crowbar_act(mob/living/user, obj/item/I)
+	if(!can_be_reached(user))
+		return TRUE
 
-		else if (istype(I, /obj/item/crowbar) && reinf && (state == WINDOW_OUT_OF_FRAME || state == WINDOW_IN_FRAME))
-			to_chat(user, "<span class='notice'>You begin to lever the window [state == WINDOW_OUT_OF_FRAME ? "into":"out of"] the frame...</span>")
-			I.play_tool_sound(src, 75)
-			if(I.use_tool(src, user, decon_speed, extra_checks = CALLBACK(src, .proc/check_state_and_anchored, state, anchored)))
-				state = (state == WINDOW_OUT_OF_FRAME ? WINDOW_IN_FRAME : WINDOW_OUT_OF_FRAME)
-				to_chat(user, "<span class='notice'>You pry the window [state == WINDOW_IN_FRAME ? "into":"out of"] the frame.</span>")
-			return
+	add_fingerprint(user)
 
-		else if(istype(I, /obj/item/wrench) && !anchored)
-			I.play_tool_sound(src, 75)
-			to_chat(user, "<span class='notice'> You begin to disassemble [src]...</span>")
-			if(I.use_tool(src, user, decon_speed, extra_checks = CALLBACK(src, .proc/check_state_and_anchored, state, anchored)))
-				var/obj/item/stack/sheet/G = new glass_type(user.loc, glass_amount)
-				G.add_fingerprint(user)
-				playsound(src, 'sound/items/Deconstruct.ogg', 50, 1)
-				to_chat(user, "<span class='notice'>You successfully disassemble [src].</span>")
-				qdel(src)
-			return
-	return ..()
+	if(reinf && (state == WINDOW_OUT_OF_FRAME || state == WINDOW_IN_FRAME))
+		to_chat(user, "<span class='notice'>You begin to lever the window [state == WINDOW_OUT_OF_FRAME ? "into":"out of"] the frame...</span>")
+		if(I.use_tool(src, user, decon_speed, volume=75, extra_checks = CALLBACK(src, .proc/check_state_and_anchored, state, anchored)))
+			state = (state == WINDOW_OUT_OF_FRAME ? WINDOW_IN_FRAME : WINDOW_OUT_OF_FRAME)
+			to_chat(user, "<span class='notice'>You pry the window [state == WINDOW_IN_FRAME ? "into":"out of"] the frame.</span>")
+		return TRUE
+	else
+		return FALSE
+
+/obj/structure/window/wrench_act(mob/living/user, obj/item/I)
+	if(!can_be_reached(user))
+		return TRUE
+
+	add_fingerprint(user)
+
+	if(!anchored)
+		to_chat(user, "<span class='notice'> You begin to disassemble [src]...</span>")
+		if(I.use_tool(src, user, decon_speed, volume=75, extra_checks = CALLBACK(src, .proc/check_state_and_anchored, state, anchored)))
+			var/obj/item/stack/sheet/G = new glass_type(user.loc, glass_amount)
+			G.add_fingerprint(user)
+			playsound(src, 'sound/items/Deconstruct.ogg', 50, 1)
+			to_chat(user, "<span class='notice'>You successfully disassemble [src].</span>")
+			qdel(src)
+		return TRUE
+	else
+		return FALSE
+
+/obj/structure/window/setAnchored(anchor_value)
+	..()
+	air_update_turf(TRUE)
+	update_nearby_icons()
+
+/obj/structure/window/proc/not_max_integrity()
+	if(obj_integrity != max_integrity)
```

Comment:
Should be a check to see if its greater as well in this check due to admin shenanigans.

Reply:
If it's greater than or equal to max integrity then this proc is never called so there's no real point.